Client/Frontend/Browser/Tab.swift

```diff
@@ -246,8 +246,8 @@ class Tab: NSObject {
             }
         }
 
-        // Don't expose localhost urls.
-        if let url = self.url, url.isLocal {
+        // When picking a display title. Tabs with a sessionData are pending a restore so show their old title.
+        if let url = self.url, url.isAboutHomeURL, sessionData == nil {
```

Comment:
This change could result in us showing some ugly localhost URLs if we make it past this `if` condition but there's no `lastTitle`. What about something like this?

```
guard let lastTitle = lastTitle, !lastTitle.isEmpty else {
    // Don't expose localhost urls.
    guard let url = self.url, !url.isLocal else { return "" }
    
    return url.displayURL?.absoluteString ?? ""
}
```